### PR TITLE
feat(aws): add IAM service-specific credentials sync

### LINKED ISSUE #1887: AWS IAM list_service_specific_credentials
AWS IAM users can have static credentials associated with them that are tied to a specific AWS service e.g. Bedrock. We should audit these in Cartography.

This would let us gain visibility on static credentials. IAM user access keys are a security antipattern so every org should monitor these credentials.

N/A

### PR DESCRIPTION
**Pull request by BABTUNA**
### Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [ ] Other (please describe):

### Summary

Add support for syncing AWS IAM service-specific credentials to Cartography. Service-specific credentials are static credentials tied to specific AWS services (e.g., Amazon Bedrock, CodeCommit) that IAM users can have associated with their accounts.

**Changes:**
- Added `AWSServiceSpecificCredential` node schema with extra label `APIKey` (enables cross-platform API key queries)
- Implemented sync functions following the standard GET → TRANSFORM → LOAD → CLEANUP pattern
- Relationships: `(AWSUser)-[SERVICE_SPECIFIC_CREDENTIAL]->(AWSServiceSpecificCredential)` and `(AWSAccount)-[RESOURCE]->(AWSServiceSpecificCredential)`
- Added unit and integration tests (including cleanup lifecycle test)
- Updated schema documentation

### Related issues or links

- Fixes #1887

### How was this tested?

- Unit tests: `uv run pytest -vvv tests/unit/cartography/intel/aws/iam/test_iam.py` — 17 passed
- Integration tests: `uv run pytest -vvv tests/integration/cartography/intel/aws/iam/test_iam_sync.py`
- Tests cover: transform logic, aggregation, edge cases (missing credential ID, NoSuchEntityException), full sync, and cleanup lifecycle

#### General
- [x] I have read the [contributing guidelines](https://cartography-cncf.github.io/cartography/dev/developer-guide.html).
- [ ] The linter passes locally (`make lint`).
- [x] I have added/updated tests that prove my fix is effective or my feature works.

#### Proof of functionality
- [x] New or updated unit/integration tests.

#### If you are adding or modifying a synced entity
- [ ] Included Cartography sync logs from a real environment demonstrating successful synchronization of the new/modified entity.

#### If you are changing a node or relationship
- [x] Updated the [schema documentation](https://github.com/cartography-cncf/cartography/tree/master/docs/root/modules).
- [ ] Updated the [schema README](https://github.com/cartography-cncf/cartography/blob/master/docs/schema/README.md).

#### If you are implementing a new intel module
- [x] Used the NodeSchema [data model](https://cartography-cncf.github.io/cartography/dev/writing-intel-modules.html#defining-a-node).

### Notes for reviewers

- This follows the same pattern as `AccountAccessKey` (access keys sync)
- Uses direct API call (`list_service_specific_credentials`) instead of paginator because this API doesn't support boto3 paginators
- `NoSuchEntityException` is handled gracefully for users deleted between Neo4j query and AWS API call

### CODE REVIEW

File: cartography/intel/aws/iam.py [hunk 1]
```diff
@@ -433,6 +436,44 @@ def get_account_access_key_data(
...
+) -> list[dict[str, Any]]:
+    client = create_boto3_client(boto3_session, "iam")
+    try:
+        response = client.list_service_specific_credentials(UserName=username)
+    except client.exceptions.NoSuchEntityException:
+        logger.warning(
+            "Could not get service-specific credentials for user %s due to NoSuchEntityException; skipping.",
+            username,
+        )
+        return []
```
[the remark this answers is not in the document]
Comment (BABTUNA) [reply cartography/intel/aws/iam.py:473]: I'm pretty sure this is a intentional pattern. There's a race condition where a user can be deleted between our Neo4j query and the AWS API call. Re-raising would crash the entire IAM sync for the account since sync() has no try/except. All 10 other NoSuchEntityException catches in this file do the same thing (log + return safe default).

**Comment by BABTUNA** [comment]
image
image
image
some screenshots of new queries added

  Neo4j queries showing synced AWSServiceSpecificCredential nodes with user
  and account relationships.

**Comment by jychp** [comment]
To help with the review, please add a screenshot of the output log from your branch tested against a real environment. This makes it easier to verify the expected behavior before merging.
